### DISCUSSION TITLE
How to bypass Setup Wizard and Autoconfigure LMS?

### DISCUSSION THREAD

**Discussion by PrinceJJPablo** [LMS-Community/slimserver#887] [Q&A] [answered]
On a new installation of LMS, <IP address>:9000/settings/server/wizard.html is launched.

I would like to bypass this setup wizard so that when LMS is launched everything is already preconfigured.

I attempted to do this by modifying /var/lib/squeezeboxserver/prefs/server.prefs

I added the music directory paths, custom skin and added "wizardDone: 1" at the end of the prefs. 
I copied how the /server.prefs looked after how you would manually do it on the /wizard.html.

Unsure if this is the correct way to bypass the wizard and autoconfigure LMS but this method doesn't seem to work.
I would modify the server.prefs then trigger: systemctl --system start squeezeboxserver.service and upon going to the browser it directs me to <IP address>:9000/settings/server/wizard.html 
If I check the /server.prefs it has all the modified lines and maintains privileges and ownerships.

LMS 8.3.1
Perl 5.34.1
Fedora 36

Is it possible to bypass the wizard and if so how?

**Comment by mherger** [comment, marked as answer]
That should work if done correctly. Are you sure the file paths, permissions, formatting of the file etc. are correct?

You mention a custom skin. Would you be able to add another file? You could create `Slim/Utils/OS/Custom.pm` with the following lines:

```perl
sub initPrefs {
	my ($class, $prefs) = @_;
	$prefs->{wizardDone} = 1;
}
```

See eg. https://github.com/Logitech/slimserver/blob/feb5e256ad91cf07e58a0cee1a49392b21c7bdd6/Slim/Utils/OS/OSX.pm#L84

**Reply by PrinceJJPablo** [reply]
It was done incorrectly, thank you for your response. 

Modifying /var/lib/squeezeboxserver/prefs/server.prefs does allow you to make configurations and adding `wizardDone: 1` at the end of the file allows you to skip the wizard. 

My issue was on my server.prefs line 355 to do with the playlist directory.
My script wrote line 335 `playlistdir: - /storage/music/playlist`
Instead of line 335: `playlistdir: /storage/music/playlist`

Does LMS detect there is an issue with the server.prefs? since even with the wrong playlist directory, it still has `wizardDone: 1` but this line was ignored and LMS continues to display the wizard with `wizardDone: 1`?

**Reply by michaelherger** [reply]
I'm not sure, tbh, how it's dealing with small syntactical errors in the file. But your experience makes me believe it ignores the rest of the file or all of it.

**Comment by gharris999** [comment]
The server.prefs file is in yaml format, isn't it?  Perhaps LMS is invoking some kind of perl yaml syntax checking on the server.prefs file and calling up the wizard if it fails.

You could perform relaxed yaml syntax checking using yamllint to insure you have a valid server.prefs file.  Normally, yamllint objects to indentation errors and long lines in server.prefs.  But you can relax those checks with:

yamllint -d "{extends: relaxed, rules: {line-length: {max: 120}}}" server.prefs

In your case, with the spurious "-" in the playlistdir path, yamllint detects the error, throwing up:

server.prefs
  1670:14   error    syntax error: sequence entries are not allowed here (syntax)

**Reply by marioqxx** [reply]
I know this is old. In case someone is looking for such a "script", here it is: https://github.com/marioqxx/lms. And the discussion is here: #902.
